tests/unit/test_onestream.py

```diff
@@ -0,0 +1,464 @@
+"""Tests for the OneStream class."""
+
+import json
+from unittest.mock import MagicMock, patch
+
+import pandas as pd
+from pydantic import ValidationError
+import pytest
+import requests
+
+from viadot.sources import OneStream
+from viadot.sources.onestream import OneStreamCredentials
+
+
+DUMMY_CREDS = {
+    "api_token": "test_api_token_123456",  # pragma: allowlist secret
+}
+
+SAMPLE_ADAPTER_RESPONSE = {
+    "Results": [
+        {"ID": 1, "Entity": "Entity1", "Account": "Account1", "Amount": 1000.50},
+        {"ID": 2, "Entity": "Entity2", "Account": "Account2", "Amount": 2000.75},
+        {"ID": 3, "Entity": "Entity1", "Account": "Account3", "Amount": 1500.25},
+    ]
+}
+
+SAMPLE_SQL_RESPONSE = {
+    "Results": [
+        {"UserName": "admin", "FullName": "Administrator", "IsActive": True},
+        {"UserName": "user1", "FullName": "User One", "IsActive": True},
+        {"UserName": "user2", "FullName": "User Two", "IsActive": False},
+    ]
+}
+
+SAMPLE_CUSTOM_SUBST_VARS = {
+    "prm_entity": ["Entity1", "Entity2"],
+    "prm_scenario": ["Actual", "Budget"],
+}
+
+
+@pytest.fixture
+def onestream_credentials():
+    """Sample OneStream credentials for testing."""
+    return DUMMY_CREDS
+
+
+# TODO: Check if this is enough
+@pytest.fixture
+def onestream_instance(onestream_credentials):
+    """Create OneStream instance with test credentials."""
+    return OneStream(
+        server_url="https://test.onestream.com",
+        application="TestApp",
+        credentials=onestream_credentials,
+    )
+
+
+def test_valid_credentials():
+    """Test creating valid OneStreamCredentials."""
+    creds = OneStreamCredentials(api_token="test_token")  # noqa: S106
+    assert creds.api_token == "test_token"  # noqa: S105
+
+
+def test_missing_api_token():
+    """Test OneStreamCredentials with missing api_token."""
+    with pytest.raises(ValueError, match="api_token"):
+        OneStreamCredentials()
+
+
+def test_credentials_from_dict():
+    """Test creating credentials from dictionary."""
+    creds_dict = {"api_token": "dict_token"}
+    creds = OneStreamCredentials(**creds_dict)
+    assert creds.api_token == "dict_token"  # noqa: S105
+
+
+@patch("viadot.sources.onestream.get_source_credentials")
+def test_init_with_credentials_dict(mock_get_creds, onestream_credentials):
+    """Test initialization with credentials dictionary."""
+    mock_get_creds.return_value = None
+    onestream = OneStream(
+        server_url="https://test.onestream.com",
+        application="TestApp",
+        credentials=onestream_credentials,
+    )
+    assert onestream.credentials == onestream_credentials
+    assert onestream.server_url == "https://test.onestream.com"
+    assert onestream.application == "TestApp"
+    assert onestream.api_token == "test_api_token_123456"  # noqa: S105
+
+
+@patch("viadot.sources.onestream.get_source_credentials")
+def test_init_with_config_key(mock_get_creds, onestream_credentials):
```

Comment:
This isn't actually testing what it says it does, since the class doesn't use `config_key` at all and yet this test passes. This is because the return value is hardcoded to what you expect to see. So you should either rewrite this test or remove it (I don't think config key matters much currently so it's ok in this case).

BTW the way to minimize the possibility of writing such bugged tests is to use the TDD way of making sure that the test is failing first, and only then making it pass after adding the required functionality (it's even more important with agents BTW since otherwise they quite commonly make this mistake, and such bugs are quite hard to catch but can make debugging things later on a nightmare).